### PR TITLE
[PTMIA062020] Arianna Valle

### PR DESCRIPTION
**Pull request by ariannavalle**

### CODE REVIEW

**Comment by ariannavalle** [comment]
The bonus iterations were very challenging, it took me hours to come up with working solutions. That being said, working through them was really fun. I really liked working with a bigger data set, it felt like a "real world" task. It's nice because it shows us that the data is not always going to be pretty and come in the way we want and prepares us to find ways to manipulate data based on requirements.

File: src/movies.js [hunk 1]
```diff
@@ -1,16 +1,106 @@
 // Iteration 1: All directors? - Get the array of all directors.
+let getAllDirectors = (movies) => {
+	let directors = movies.map((movie) => movie.director);
+	return directors;
+};
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
+let removeDuplicates = (movies) => {
+	let directors = getAllDirectors(movies);
+	let uniqueDirectors = directors.filter(
```
Comment (sandrabosk) [inline src/movies.js:10]: Great approach to solving this!!

File: src/movies.js [hunk 2]
```diff
@@ -1,16 +1,106 @@
...
+};
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+let ratesAverage = (movies) => {
+	if (!movies.length) return 0;
+	let sumOfRatings = movies.reduce((accumulator, current) => {
+		if (!current.rate) current.rate = 0;
+		return accumulator + current.rate;}, 0);
+	let avgRatings = sumOfRatings / movies.length;
+	avgRatings = Number(avgRatings.toFixed(2));
```
Comment (sandrabosk) [inline src/movies.js:31]: Can you think of some other ways to turn string to a number?

File: src/movies.js [hunk 3]
```diff
@@ -1,16 +1,106 @@
...
+	return avgRatings;
+};
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
+let dramaMoviesRate = (movies) => {
+	if (!movies.length) return 0;
+	let dramaMovies = movies.filter((movie) => movie.genre.includes("Drama"));
+	if (!dramaMovies.length) return 0;
+	return ratesAverage(dramaMovies);
```
Comment (sandrabosk) [inline src/movies.js:40]: Code reusability on point! Great.

File: src/movies.js [hunk 4]
```diff
@@ -1,16 +1,106 @@
...
+	let replaceWithMinutes = movies.map((movie, index) => {
+		return {...movie, duration : time[index]};
+	});
+
+	return replaceWithMinutes;
+};
 
 // BONUS - Iteration 8: Best yearly rate average - Best yearly rate average
+
+let bestYearAvg = (movies) => {
```
Comment (sandrabosk) [inline src/movies.js:89]: Your solution works, and I am impressed with the approach you took and that you didn't give up on bonus section. I will leave below a bit shorter and cleaner version, just so you have it as a reference:
```js

function bestYearAvg(lotsOfMovies) {
  if (!lotsOfMovies.length) return null;

  let masterObject = {};

  lotsOfMovies.forEach(eachMovie => {
    if (!masterObject[eachMovie.year]) {
      masterObject[eachMovie.year] = [eachMovie];
    } else {
      masterObject[eachMovie.year].push(eachMovie);
    }
  });

  let highest = 0;
  let theActualYear;
  for (let theYear in masterObject) {
    if (ratesAverage(masterObject[theYear]) > highest) {
      highest = ratesAverage(masterObject[theYear]);
      theActualYear = theYear;
    }
  }
  return `The best year was ${theActualYear} with an average rate of ${highest}`;
}
```

**Comment by sandrabosk** [comment]
First, shoutout to making this PR with multiple commits! Way to go.
Second, your solutions are really good and you are definitely on the right track. Your code looks clean, following the best practices. Keep it up!!

**Pull request closed without merging**